src/pybind/mgr/cephadm/schedule.py

```diff
@@ -433,3 +438,14 @@ def get_candidates(self) -> List[DaemonPlacement]:
         final = sorted(ls)
         random.Random(seed).shuffle(final)
         return ls
+
+    def remove_non_maintenance_unreachable_candidates(self, candidates: List[DaemonPlacement]) -> List[DaemonPlacement]:
```

Comment:
I wasn't able to figure out what hosts are considered to be in maintenance. So wasn't able to also figure out why we include hosts in maintenance mode as candidates.

Reply:
we don't want to re-schedule daemons which are placed on the nodes in maintenance mode ...